src/commands/advice.rs

```diff
@@ -1,26 +1,25 @@
-// use serenity::framework::standard::{macros::command, Args, CommandResult};
-// use serenity::model::prelude::*;
-// use serenity::prelude::*;
+use serde::Deserialize;
+use serenity::framework::standard::{macros::command, CommandResult};
+use serenity::model::prelude::*;
+use serenity::prelude::*;
 
-// #[command]
-// async fn advice(ctx: &Context, msg: &Message) -> CommandResult {
-//     let endpoint = "https://api.adviceslip.com/advice";
-//     let result_text = reqwest::blocking::get(endpoint)?.json()?;
-//     // let result = serde_json::from_str(&result_text).expect("json from string failed");
-//     let results = format!("{:?}", result_text);
+#[derive(Deserialize)]
+struct Slip {
+    id: i32,
+    advice: String,
+}
 
-//     let _ = msg.channel_id.say(&ctx.http, results).await;
-//     Ok(())
-// }
+#[derive(Deserialize)]
+struct Advice {
+    slip: Slip,
+}
 
-// #[command]
-// async fn advice_id(_ctx: &Context, _msg: &Message, _args: Args) -> CommandResult {
-//     let _endpoint = "https://api.adviceslip.com/advice/{slip_id}";
-//     Ok(())
-// }
+#[command]
+async fn advice(ctx: &Context, msg: &Message) -> CommandResult {
+    let endpoint = "https://api.adviceslip.com/advice";
+    let result_text = reqwest::get(endpoint).await?.json::<Advice>().await?;
```

Comment:
result_text seems like a bad name after it's been parsed to a real struct.

Reply:
Lol. Good point.

Reply:
Fixed.